src/ast/passes/semantic_analyser.cpp

```diff
@@ -37,9 +37,53 @@ static const std::map<std::string, std::tuple<size_t, bool>> &getIntcasts()
   return intcasts;
 }
 
+template <typename T>
+static std::pair<T, T> getIntTypeRange(const SizedType &ty)
```

Comment:
Right now we're checking `IsSigned()` before the call to `getIntTypeRange` to determine which template parameter to use, and then checking it again inside this function to find the actual range. Also, the function `getIntTypeRange<int64_t>` sounds like it might return the range specifically for 64-bit integers (and not 8, 16, 32 bit ints).

How about just having two functions that don't have to check `IsSigned` inside them:
```
std::pair<int64_t, int64_t> getIntTypeRange(...)
std::pair<uint64_t, uint64_t> getUIntTypeRange(...)
```

Reply:
I like it. Good idea 💡